x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -598,6 +608,50 @@ export class AlertsClient {
     }
   }
 
+  public async find<Params extends AlertTypeParams = never>({
+    query,
+    aggs,
+    _source,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    track_total_hits,
+    size,
+    index,
+  }: {
+    query?: object | undefined;
+    aggs?: object | undefined;
+    index: string | undefined;
+    track_total_hits?: boolean | undefined;
+    _source?: string[] | undefined;
+    size?: number | undefined;
+  }) {
```

Comment:
How does this `find()` method compare with `RuleDataClient.getReader().search()`, when should we use which one, etc? Do we even need `RuleDataClient.getReader()` in the current form?